collect_app/src/main/java/org/odk/collect/android/widgets/RatingWidget.java

```diff
@@ -14,116 +14,92 @@
-import android.widget.GridLayout;
-import android.widget.ImageButton;
+import android.view.View;
+import android.widget.RatingBar;
 
 import org.javarosa.core.model.RangeQuestion;
 import org.javarosa.core.model.data.IAnswerData;
 import org.javarosa.core.model.data.IntegerData;
-import org.odk.collect.android.R;
+import org.javarosa.form.api.FormEntryPrompt;
+import org.odk.collect.android.databinding.RatingWidgetAnswerBinding;
 import org.odk.collect.android.formentry.questions.QuestionDetails;
-import org.odk.collect.android.formentry.questions.WidgetViewUtils;
+import org.odk.collect.android.utilities.UiUtils;
 
+@SuppressLint("ViewConstructor")
 public class RatingWidget extends QuestionWidget {
 
     public static final int ASSUMED_TOTAL_MARGIN_AROUND_WIDGET = 40;
 
-    final GridLayout gridLayout;
-    Integer answer;
+    private RatingBar ratingBar;
 
     public RatingWidget(Context context, QuestionDetails questionDetails) {
         super(context, questionDetails);
+    }
 
-        RangeQuestion rangeQuestion = (RangeQuestion) getFormEntryPrompt().getQuestion();
-        int numberOfStars = rangeQuestion.getRangeEnd().intValue();
-        int columns = calculateColumns();
-        int rows = (int) Math.ceil((double) numberOfStars / columns);
+    @Override
+    protected View onCreateAnswerView(Context context, FormEntryPrompt prompt, int answerFontSize) {
+        RatingWidgetAnswerBinding binding = RatingWidgetAnswerBinding.inflate(((Activity) context).getLayoutInflater());
+        View answerView = binding.getRoot();
 
-        gridLayout = new GridLayout(context);
-        gridLayout.setLayoutParams(new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT));
 
-        renderGrid(context, numberOfStars, columns, rows);
+        RangeQuestion rangeQuestion = (RangeQuestion) prompt.getQuestion();
+        int numberOfStars = rangeQuestion.getRangeEnd().intValue();
+        int columns = calculateColumns(48);
 
-        String answerText = questionDetails.getPrompt().getAnswerText();
-        if (answerText != null) {
-            answer = Integer.parseInt(answerText);
-            for (int i = 0; i < answer; i++) {
-                ImageButton button = (ImageButton) gridLayout.getChildAt(i);
-                button.setImageResource(R.drawable.ic_star);
+        int rows = (int) Math.ceil((double) numberOfStars / columns);
+
+        if (rows == 1) {
+            binding.ratingBar.setVisibility(VISIBLE);
+            this.ratingBar = binding.ratingBar;
+        } else {
+            binding.ratingBar.setVisibility(GONE);
+            columns = calculateColumns(36);
+            rows = (int) Math.ceil((double) numberOfStars / columns);
+            if (rows == 1) {
+                binding.ratingBar2.setVisibility(VISIBLE);
+                this.ratingBar = binding.ratingBar2;
+            } else {
+                binding.ratingBar3.setVisibility(VISIBLE);
+                this.ratingBar = binding.ratingBar3;
```

Comment:
I have three `Ratingbar` with different styles In the XML file. All of thems visibility is turned off by default. This part of code checks which `Ratingbar` needs to be displayed depending on the number of stars and screen dimensions.